pkg/cli/verifier/connectivity_pod_to_pod.go

```diff
@@ -0,0 +1,68 @@
+package verifier
+
+import (
+	"fmt"
+	"io"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+)
+
+var _ Verifier = (*PodConnectivityVerifier)(nil)
+
+// PodConnectivityVerifier implements the Verifier interface for pod connectivity
+type PodConnectivityVerifier struct {
+	stdout      io.Writer
+	stderr      io.Writer
+	kubeClient  kubernetes.Interface
+	srcPod      types.NamespacedName
+	dstPod      types.NamespacedName
+	appProtocol string
+	meshName    string
+}
+
+// NewPodConnectivityVerifier implements verification for pod connectivity
+func NewPodConnectivityVerifier(stdout io.Writer, stderr io.Writer, kubeClient kubernetes.Interface,
+	srcPod types.NamespacedName, dstPod types.NamespacedName, appProtocol string, meshName string) Verifier {
+	return &PodConnectivityVerifier{
+		stdout:      stdout,
+		stderr:      stderr,
+		kubeClient:  kubeClient,
+		srcPod:      srcPod,
+		dstPod:      dstPod,
+		appProtocol: appProtocol,
+		meshName:    meshName,
+	}
+}
+
+// Run executes the pod connectivity verifier
+func (v *PodConnectivityVerifier) Run() VerificationResult {
+	result := VerificationResult{
+		Context: fmt.Sprintf("Verify if pod %q can access pod %q for app protocol %q", v.srcPod, v.dstPod, v.appProtocol),
+	}
+
+	verifiers := []Verifier{
+		// ---
+		// Verify prerequisites
+		//
+		// Namespace monitor verification
+		NewNamespaceMonitorVerifier(v.stdout, v.stderr, v.kubeClient, v.srcPod.Namespace, v.meshName),
+		NewNamespaceMonitorVerifier(v.stdout, v.stderr, v.kubeClient, v.dstPod.Namespace, v.meshName),
+	}
+
+	for _, verification := range verifiers {
+		res := verification.Run()
+		if res.Status == Failure {
+			result.Status = Failure
+			result.Reason = "A verification step failed"
+			result.Suggestion = "Please follow the suggestions listed in the failed steps below to resolve the issue"
+		}
+		result.NestedResults = append(result.NestedResults, &res)
+	}
+
+	if result.Status != Failure && result.Status != Unknown {
```

Comment:
I don't think result.Status == Unknown is ever currently possible?

Reply:
At the moment yes, but it needs to be used when the verifier cannot determine the result, for e.g. when an API call fails. As I add more verifications, this will become evident.